pkg/minikube/performance/start.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2017 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package performance
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	runs = 1
+	// For testing
+	collectTimeMinikubeStart = timeMinikubeStart
+)
+
+// CompareMinikubeStart compares the time to run `minikube start` between two minikube binaries
+func CompareMinikubeStart(ctx context.Context, out io.Writer, binaries []*Binary) error {
+	durations, err := collectTimes(ctx, binaries)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(out, "Old binary: %v\nNew binary: %v\nAverage Old: %f\nAverage New: %f\n", durations[0], durations[1], average(durations[0]), average(durations[1]))
+	return nil
+}
+
+func collectTimes(ctx context.Context, binaries []*Binary) ([][]float64, error) {
+	durations := make([][]float64, len(binaries))
+	for i := range durations {
+		durations[i] = make([]float64, runs)
+	}
+
+	for r := 0; r < runs; r++ {
+		log.Printf("Executing run %d...", r)
+		for index, binary := range binaries {
+			duration, err := collectTimeMinikubeStart(ctx, binary)
+			if err != nil {
+				return nil, errors.Wrapf(err, "timing run %d with %s", r, binary.path)
+			}
+			durations[index][r] = duration
+		}
+	}
+
+	return durations, nil
+}
+
+func average(array []float64) float64 {
```

Comment:
Use another name rather than `array`: this term is reserved for fixed-size slices in Go lang.

Reply:
done